hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -317,6 +318,13 @@ private TransactionBody handleUserTransaction(
             } else {
                 feeAccumulator.charge(payer, fees);
                 try {
+                    final var storeFactory = new ReadableStoreFactory(state);
+                    final var accountStore = storeFactory.getStore(ReadableAccountStore.class);
+                    final var payerAccount = accountStore.getAccountById(payer);
+                    if (payerAccount != null && payerAccount.deleted()) {
```

Comment:
This check might be changed to other lines when #8224 is completed.

Reply:
This looks like a workaround. We should not use the payer from pre-handle, but re-read it from state in handle and recheck its state. Let's fix it correctly right away. You can go ahead and do it or file an issue and assign it to me.

Reply:
Updated with getting payer (AccountID) from transaction. payerAccount was from state in handle on line 323.

Reply:
Should we also be checking for other special cases? Specifically, deleted contracts and expired accounts

Reply:
This PAYER_ACCOUNT_DELETED is specific to payerAccount. If we have to check deleted contracts and expired account here, we can address it in another issue.  I will let @netopyr comment on deleted contracts and expired account.

Reply:
There is a whole facility for expired items that needs to be added still.  It would be outside the scope of this PR.

Reply:
Is there more to do to detect a deleted contract?

I have only a very basic understanding of how expiration will work. Let's discuss it today.